packages/components/pie-avatar/src/index.ts

```diff
@@ -1,22 +1,105 @@
-import { RtlMixin, safeCustomElement } from '@justeattakeaway/pie-webc-core';
+import { RtlMixin, safeCustomElement, validPropertyValues } from '@justeattakeaway/pie-webc-core';
 
+import { property } from 'lit/decorators.js';
 import styles from './avatar.scss?inline';
-import { type AvatarProps } from './defs';
+import {
+    type AvatarProps, defaultProps, tags,
+} from './defs';
 
 // Valid values available to consumers
 export * from './defs';
 
 const componentSelector = 'pie-avatar';
 
+type Initials = {
+    visual: string,
+    screenreader: string
+}
+
 /**
  * @tagname pie-avatar
  */
 @safeCustomElement('pie-avatar')
 export class PieAvatar extends RtlMixin(PieElement) implements AvatarProps {
+    @property({ type: String })
+    @validPropertyValues(componentSelector, tags, defaultProps.tag)
+    public tag = defaultProps.tag; // access modifier
+
+    @property({ type: String })
+    public label: AvatarProps['label'];
+
+    // Attempts to extract initials from the label string. If the label is not provided or is invalid, it returns null.
+    private getInitials (name: string): Initials | null {
+        try {
+            if (!name || typeof name !== 'string' || name.trim().length === 0) {
+                return null;
+            }
+
+            const nameSplit: string[] = name.trim().replace(/-/g, ' ').split(/\s+/); // [Ada, Lovelace]
+            const initials: string[] = nameSplit.slice(0, 2).map((word) => word[0].toUpperCase()); // [A, L]
+
+            if (initials.length === 0) {
+                return null;
+            }
+
+            return {
+                visual: initials.join(''),
+                screenreader: initials.join(', '), // joins the two words by comma so initials are correctly pronounced by screenreaders
+            };
+        } catch (error) {
+            return null;
+        }
+    }
+
+    // Renders the initials both for visual display and for screen readers.
+    private renderInitials (initials: Initials): TemplateResult {
+        return html`
+            <span aria-hidden="true" data-test-id="pie-avatar-initials-visual">${initials.visual}</span>
+            <span class="c-avatar-visuallyHidden" data-test-id="pie-avatar-initials-screenreader">${initials.screenreader}</span>
+        `;
+    }
+
+    // Renders the icon (placeholder span for now)
+    private renderIcon (): TemplateResult {
+        return html`<span data-test-id="pie-avatar-icon" class="c-avatar-placeholder">Icon Placeholder</span>`;
+    }
+
+    // Renders the inner content of the avatar such as initials, an icon or an image
+    // This is a getter because the value is computed based on properties
+    private get avatarContent (): TemplateResult {
+        // TODO: handle unauthenticated and src here
+
+        if (this.label) {
+            const initials = this.getInitials(this.label);
+            if (initials) {
+                return this.renderInitials(initials);
+            }
```

Comment:
do you think we can encapsulate the logic of getting getInitials in renderInitials?
this way, the method will be responsible for everything and might be easier to read. 

```
    private renderInitials (): TemplateResult {
   const initials = this.getInitials(this.label);
            if (!initials) return nothing  
        return html`
            <span aria-hidden="true" data-test-id="pie-avatar-initials-visual">${initials.visual}</span>
            <span class="c-avatar-visuallyHidden" data-test-id="pie-avatar-initials-screenreader">${initials.screenreader}</span>
        `;
    }
```

Reply:
I think by the end of it, this would lead to more complex code in this method. We will keep it as-is for now and refactor later on if it seems necessary :)